udata/harvest/backends/dcat.py

```diff
@@ -58,10 +63,28 @@ def initialize(self):
         '''List all datasets for a given ...'''
         fmt = self.get_format()
         graphs = self.parse_graph(self.source.url, fmt)
-        self.job.data = {
-            'graphs': [graph.serialize(format=fmt, indent=None) for graph in graphs],
-            'format': fmt,
-        }
+
+        self.job.data = { 'format': fmt }
+
+        serialized_graphs = [graph.serialize(format=fmt, indent=None) for graph in graphs]
+
+        # The official MongoDB document size in 16MB. The default value here is 15MB to account for other fields in the document (and for difference between * 1024 vs * 1000).
+        max_harvest_graph_size_in_mongo = current_app.config.get('HARVEST_MAX_CATALOG_SIZE_IN_MONGO')
+        if max_harvest_graph_size_in_mongo is None:
+            max_harvest_graph_size_in_mongo = 15 * 1000 * 1000
+
+        bucket = current_app.config.get('HARVEST_GRAPHS_S3_BUCKET')
+
+        if bucket is not None and sum([len(g.encode('utf-8')) for g in serialized_graphs]) >= max_harvest_graph_size_in_mongo:
+            # TODO: we could store each page in independant files to allow downloading only the require page in
+            # subsequent jobs. (less data to download in each job)
+            filename = f'harvest_{self.job.id}_{date.today()}.json'
+
+            store_as_json(bucket, filename, serialized_graphs)
```

Comment:
can you modify this function to add a `HARVEST_GRAPHS_S3_PATH` in it. Indeed, we can use a bucket for multiple things other than harvester. We can imagine to store all of this in an `/harvest` path for instance.

Reply:
Done in af58134a9